distribution/metadata/diff_metadata_service.go

```diff
@@ -0,0 +1,137 @@
+package metadata
+
+import (
+	"encoding/json"
+
+	"github.com/docker/distribution/digest"
+	"github.com/docker/docker/layer"
+)
+
+// DiffMetadataService maps layer IDs to a set of known metadata for
+// the layer.
+type DiffMetadataService struct {
```

Comment:
@aaronlehmann let me just give you my full support on naming issues. It is important. Always.
However, since it's not user-facing, I agree it is less critical: we can always rename it later, but let's not forget to do it.